packages/lexicon/src/types.ts

```diff
@@ -229,26 +234,68 @@ export type LexRecord = z.infer<typeof lexRecord>
 // core
 // =
 
-export const lexUserType = z.discriminatedUnion('type', [
-  lexRecord,
-
-  lexXrpcQuery,
-  lexXrpcProcedure,
-  lexXrpcSubscription,
-
-  lexBlob,
+export const lexUserType = z.custom<
+  | LexRecord
+  | LexXrpcQuery
+  | LexXrpcProcedure
+  | LexXrpcSubscription
+  | LexBlob
+  | LexArray
+  | LexToken
+  | LexObject
+  | LexBoolean
+  | LexInteger
+  | LexString
+  | LexBytes
+  | LexCidLink
+  | LexUnknown
+>(
+  (val) => {
+    if (!val || typeof val !== 'object') {
+      return
+    }
 
-  lexArray,
-  lexToken,
-  lexObject,
+    if (val['type'] === undefined) {
+      return
+    }
```

Comment:
Just making sure I understand—these plain `return` statements get turned into errors by zod, is that right?

Reply:
You're right, the error messages were less descriptive than before, I've changed it